src/main/java/org/fcrepo/audit/InternalAuditor.java

```diff
@@ -0,0 +1,260 @@
+
+import javax.annotation.PostConstruct;
+import javax.annotation.PreDestroy;
+import javax.inject.Inject;
+import javax.jcr.Node;
+import javax.jcr.PropertyType;
+import javax.jcr.RepositoryException;
+
+import org.fcrepo.kernel.identifiers.PidMinter;
+import org.fcrepo.kernel.observer.FedoraEvent;
+import org.fcrepo.kernel.services.ContainerService;
+import org.fcrepo.kernel.utils.EventType;
+import org.fcrepo.mint.UUIDPathMinter;
+
+import org.modeshape.jcr.api.Repository;
+import org.modeshape.jcr.api.Session;
+import org.slf4j.Logger;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.base.Function;
+import com.google.common.base.Joiner;
+import com.google.common.collect.Iterables;
+import com.google.common.eventbus.EventBus;
+import com.google.common.eventbus.Subscribe;
+
+/**
+ * @author mohideen
+ * @date 4/15/15.
+ */
+public class InternalAuditor implements Auditor {
+
+    /**
+     * Logger for this class.
+     */
+    private static final Logger LOGGER = getLogger(InternalAuditor.class);
+
+    private static final String AUDIT_CONTAINER = "fcrepo.audit.container";
+
+    private static String AUDIT_CONTAINER_LOCATION;
+
+    @Inject
+    private EventBus eventBus;
+
+    @Inject
+    private Repository repository;
+
+    @Inject
+    private ContainerService containerService;
+
+    private Session session;
+
+    private static final PidMinter pidMinter =  new UUIDPathMinter();
+
+    /**
+     * Register with the EventBus to receive events.
+     */
+    @PostConstruct
+    public void register() throws RepositoryException {
+        AUDIT_CONTAINER_LOCATION = System.getProperty(AUDIT_CONTAINER);
+        if (AUDIT_CONTAINER_LOCATION != null) {
+            LOGGER.info("Initializing: {}", this.getClass().getCanonicalName());
+            eventBus.register(this);
+            if (!AUDIT_CONTAINER_LOCATION.startsWith("/")) {
+                AUDIT_CONTAINER_LOCATION = "/" + AUDIT_CONTAINER_LOCATION;
+            }
+            session = repository.login();
+            containerService.findOrCreate(session, AUDIT_CONTAINER_LOCATION);
+            session.save();
+        } else {
+            LOGGER.warn("Cannot Initialize: {}", this.getClass().getCanonicalName());
+            LOGGER.warn("System property not found: " + AUDIT_CONTAINER);
+        }
+    }
+
+    /**
+     * Fedora internal events are received and processed by this method.
+     *
+     * @param event
+     *        The {@link FedoraEvent} to record.
+     * @throws RepositoryException
+     */
+    @Subscribe
+    public void recordEvent(final FedoraEvent event) throws RepositoryException, IOException {
+        LOGGER.debug("Event detected: {} {}", event.getUserID(), event.getPath());
+        boolean isParentNodeLastModifiedEvent = false;
+        final String eventType = getEventURIs(event.getTypes());
+        final String properties = Joiner.on(',').join(event.getProperties());
```

Comment:
Wouldn't it be easier to just get the properties as a Set and iterate over them until you found a property that wasn't lastModified or lastModifiedBy?  Or maybe use Collections2.filter() to get other properties?